tests/unit/scoreBoard.spec.js

```diff
@@ -0,0 +1,185 @@
+
+        // Second throw
+        const pinsFallen2 = Math.floor(Math.random() * nbPins)
+        wrapper.vm.registerThrow(pn, i + 1, 2, pinsFallen2)
+        expect(wrapper.vm.globalScore[pn][i].throws[1]).toBe(pinsFallen2)
+
+        // Third throw
+        if (i === nbTurn - 1) {
+          const pinsFallen3 = Math.floor(Math.random() * nbPins)
+          wrapper.vm.registerThrow(pn, i + 1, 3, pinsFallen3)
+          expect(wrapper.vm.globalScore[pn][i].throws[2]).toBe(pinsFallen3)
+        }
+      }
+    })
+  })
+  test('curentPlayer props', async () => {
+    // create games params
+    const nbTurn = 20
+    const nbPins = 100
+    const playerNames = ['tata', 'tutu', 'toto']
+    const curentplayerIndex = 1
+    const players = []
+
+    playerNames.forEach(pn => {
+      players.push(new Player(pn))
+    })
+
+    const gameParam = new GameParam(players, nbTurn, nbPins)
+    const wrapperWithPlayerProps = mountComponent(gameParam, playerNames[curentplayerIndex], null)
+
+    expect(wrapperWithPlayerProps.find('#playerNames .curent').text()).toContain(playerNames[curentplayerIndex])
+
+    const wrapperWithNoPlayerProps = mountComponent(gameParam, null, null)
+    expect(wrapperWithNoPlayerProps.find('#playerNames .curent').exists()).toBe(false)
+  })
+  test('curentTurn props', async () => {
+    // create games params
+    const nbTurn = 20
+    const nbPins = 100
+    const playerNames = ['tata', 'tutu', 'toto']
+    const curentTurn = 69420
+    const players = []
+
+    playerNames.forEach(pn => {
+      players.push(new Player(pn))
+    })
+
+    const gameParam = new GameParam(players, nbTurn, nbPins)
+    const wrapperWithCurTurnProps = mountComponent(gameParam, null, curentTurn)
+
+    expect(wrapperWithCurTurnProps.find('#playerNames #curentTurnNumber').text()).toContain(curentTurn)
+
+    const wrapperWithNoCurTurnProps = mountComponent(gameParam, null, null)
+    expect(wrapperWithNoCurTurnProps.find('#playerNames #curentTurnNumber').text()).toContain('')
+  })
+  test('Classic game, wrong insertions', async () => {
+    // create games params
+    const nbTurn = 20
+    const nbPins = 100
+    const playerNames = ['tata', 'tutu', 'toto']
+    const players = []
+
+    playerNames.forEach(pn => {
+      players.push(new Player(pn))
+    })
+
+    const gameParam = new GameParam(players, nbTurn, nbPins)
+    const wrapper = mountComponent(gameParam)
+
+    // Check comp initialisation
+    playerNames.forEach(pn => {
+      for (let i = 0; i < nbTurn; i++) {
+        const pinsFallen1 = Math.floor(Math.random() * nbPins)
+
+        // Wrong player
+        try {
+          wrapper.vm.registerThrow('IDONTEXIST', i + 1, 1, pinsFallen1)
+        } catch (error) {
+          expect(error.toString()).toBe('Error: Player not found')
+        }
+
+        // Wrong turn (3, but too low)
+        try {
+          wrapper.vm.registerThrow(pn, -1, 1, pinsFallen1)
```

Comment:
Utilises plutôt `expect($lambda).toThrow($message)`

Reply:
marche pas
https://stackoverflow.com/questions/59228847/how-to-test-an-error-of-a-method-in-a-vue-js-component-using-jest
![image](https://user-images.githubusercontent.com/59999324/103577994-18bdec80-4ed6-11eb-9aa0-8f67c2f22755.png)
![image](https://user-images.githubusercontent.com/59999324/103578010-21162780-4ed6-11eb-9c99-019c963758c0.png)
je vais faire ça : 
```js
    // Check comp initialisation
    playerNames.forEach(pn => {
      for (let i = 0; i < nbTurn; i++) {
        const pinsFallen1 = Math.floor(Math.random() * nbPins)

        // Wrong player
        try {
          wrapper.vm.registerThrow('tata', i + 1, 1, pinsFallen1)
          throw (Error('The test should have failed'))
        } catch (error) {
          expect(error.toString()).toBe('Error: Player not found')
        }

```

Reply:
Le `expect` + `toThrow` attends une fonction/lambda en paramètre et pas une valeur